server/utils/cache.ts

```diff
@@ -0,0 +1,49 @@
+import { IncomingMessage } from 'http';
+import NodeCache from 'node-cache';
+
+const CACHE_TTL_ONE_MINUTE = 1 * 60;
+const cache = new NodeCache({
+  stdTTL: CACHE_TTL_ONE_MINUTE,
+  forceString: true,
+});
+
+type CacheEntry = {
+  contentType: string;
+  data: string;
+};
+
+const createCacheKey = (req: IncomingMessage): string => {
+  return req.url;
+};
+
+const createClearKey = (req: IncomingMessage): string => {
```

Comment:
For createClear, the req.url will be (for example)
`url: '/v0/baseId/users/recordId',` so when it req.url.match is passed, it returns `NULL` and when it is assigned on line 21 it will error out.

I recommend the following change -- this will ensure that it is in a format that the regex expects and allows the current code to still clear the cache of previous GET requests that are associated with any non-GET requests

suggestion fix
```
const createClearKey = (req: IncomingMessage): string => {
  const req_url = new URL(req.url, `http://${req.headers.host}`);
  const urlParse = /https?:\/\/.*\/(v[^/]*)\/(app[^/]+)\/([^?/]+)[\/.+]*/;
  const [, version, baseId, tableName] = req_url.href.match(urlParse);
  const cacheKey = [version, baseId, tableName].join('/');
  return cacheKey;
};
```